### ISSUE TITLE
Firmware Upgrade issue

### ISSUE THREAD

**Issue by Drey** [cisco-intersight/terraform-provider-intersight#10]
Hi folks,

I have an issue with the intersight_firmware_upgrade resource usage. It seems like the resource must have `cifs_server` and `nfs_server` options with a `file_location` param defined even if `map_type = "www"` is used. **The working sample shown below**:

```
resource "intersight_firmware_upgrade" "sprint-firmware-update" {
  upgrade_type =  "network_upgrade"


  network_share {
    map_type = "www"

    nfs_server {
        # Need to provide fake image due to an issue
        file_location = "10.5.30.39/staticfiles/ucs-c240m5-huu-4.1.1f.iso"
    }

    cifs_server {
        # Need to provide fake image due to an issue
        file_location = "10.5.30.39/staticfiles/ucs-c240m5-huu-4.1.1f.iso" 
    }

    http_server {
      location_link = "http://10.5.30.39/staticfiles/ucs-c240m5-huu-4.1.1f.iso"
    }
  }


  server {
    object_type = "compute.RackUnit"
    moid = var.compute-sprint-srv-1
  }
}
```

**If I don't provide cifs_server & nfs_server params, terraform fails with the following error:**

 ```
===========> {"NetworkShare":{"CifsServer":{},"HttpServer":{"LocationLink":"http://10.5.30.39/staticfiles/ucs-c240m5-huu-4.1.1f.iso"},"IsPasswordSet":false,"MapType":"www","NfsServer":{},"Upgradeoption":"nw_upgrade_full"},"Server":{"Moid":"5ed12cc36176752d30f8494a","ObjectType":"compute.RackUnit"},"UpgradeType":"network_upgrade"}
2020-06-02T12:31:14.074+0300 [DEBUG] plugin.terraform-provider-intersight_v0.1.3: 2020/06/02 12:31:14 config.go:133: response Status: 400 Bad Request
<edited>
2020-06-02T12:31:14.074+0300 [DEBUG] plugin.terraform-provider-intersight_v0.1.3: 2020/06/02 12:31:14 config.go:139: response Body: {"code":"InvalidRequest","message":"Cannot set the property 'firmware.CifsServer.NetworkShare.CifsServer.FileLocation'. The property cannot be empty.","messageId":"barcelona_request_cannot_access_property_required","messageParams":{"1":"firmware.CifsServer","2":"NetworkShare.CifsServer.FileLocation"},"traceId":"NB9451f272a03fa3019650443af9804fb7"}
2020-06-02T12:31:14.074+0300 [DEBUG] plugin.terraform-provider-intersight_v0.1.3: 2020/06/02 12:31:14 config.go:141: response Status: 400 Bad Request
<edited>
2020/06/02 12:31:14 [DEBUG] intersight_firmware_upgrade.sprint-firmware-update: apply errored, but we're indicating that via the Error pointer rather than returning it: SendRequest failed. Url firmware/Upgrades Status Code: 400 Message: {"code":"InvalidRequest","message":"Cannot set the property 'firmware.CifsServer.NetworkShare.CifsServer.FileLocation'. The property cannot be empty.","messageId":"barcelona_request_cannot_access_property_required","messageParams":{"1":"firmware.CifsServer","2":"NetworkShare.CifsServer.FileLocation"},"traceId":"NB9451f272a03fa3019650443af9804fb7"}
2020/06/02 12:31:14 [ERROR] <root>: eval: *terraform.EvalApplyPost, err: SendRequest failed. Url firmware/Upgrades Status Code: 400 Message: {"code":"InvalidRequest","message":"Cannot set the property 'firmware.CifsServer.NetworkShare.CifsServer.FileLocation'. The property cannot be empty.","messageId":"barcelona_request_cannot_access_property_required","messageParams":{"1":"firmware.CifsServer","2":"NetworkShare.CifsServer.FileLocation"},"traceId":"NB9451f272a03fa3019650443af9804fb7"}
2020/06/02 12:31:14 [ERROR] <root>: eval: *terraform.EvalSequence, err: SendRequest failed. Url firmware/Upgrades Status Code: 400 Message: {"code":"InvalidRequest","message":"Cannot set the property 'firmware.CifsServer.NetworkShare.CifsServer.FileLocation'. The property cannot be empty.","messageId":"barcelona_request_cannot_access_property_required","messageParams":{"1":"firmware.CifsServer","2":"NetworkShare.CifsServer.FileLocation"},"traceId":"NB9451f272a03fa3019650443af9804fb7"}

Error: SendRequest failed. Url firmware/Upgrades Status Code: 400 Message: {"code":"InvalidRequest","message":"Cannot set the property 'firmware.CifsServer.NetworkShare.CifsServer.FileLocation'. The property cannot be empty.","messageId":"barcelona_request_cannot_access_property_required","messageParams":{"1":"firmware.CifsServer","2":"NetworkShare.CifsServer.FileLocation"},"traceId":"NB9451f272a03fa3019650443af9804fb7"}

  on fimrware_update.tf line 1, in resource "intersight_firmware_upgrade" "sprint-firmware-update":
   1: resource "intersight_firmware_upgrade" "sprint-firmware-update" {


2020-06-02T12:31:14.098+0300 [DEBUG] plugin: plugin process exited: path=/Users/idrey/.terraform.d/plugins/darwin_amd64/terraform-provider-intersight_v0.1.3 pid=83326
2020-06-02T12:31:14.098+0300 [DEBUG] plugin: plugin exited
```

I think the error occurs because Intersight firmware/Upgrades API request always has `"NfsServer": {}` and `"CifsServer": {}` params and it doesn't matter if nfs_server and cifs_server options were defined or not. It describes why it works if I provide fake nfs & cifs file locations.

**Comment by vvb** [comment]
@Drey Thank you for reporting this issue. Certainly looks like an issue. We are checking this out.

**Comment by code-lucidal58** [comment]
@Drey The issue is resolved, and the fix will go out in next release.

**Comment by vvb** [comment]
available in v0.1.3
